src/cloudai/systems/standalone/standalone_system.py

```
@@ -78,6 +78,13 @@ def kill(self, job: BaseJob) -> None:
         Args:
             job (BaseJob): The job to be terminated.
         """
+        if str(job.id).strip() == "0":
+            logging.warning(
+                "Skipping termination for standalone job %s because it does not reference a launched process.",
+                job.id,
+            )
+            return
+
         cmd = f"kill -9 {job.id}"
         logging.debug(f"Executing termination command for job {job.id}: {cmd}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)